src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripDesigner.cs

```diff
@@ -1395,14 +1413,20 @@ public override void Initialize(IComponent component)
     {
         base.Initialize(component);
         AutoResizeHandles = true;
-        if (TryGetService(out _host))
+        if (ComponentChangeService is not null)
         {
-            _componentChangeService = (IComponentChangeService)_host.GetService(typeof(IComponentChangeService));
+            ComponentChangeService.ComponentAdding += ComponentChangeSvc_ComponentAdding;
+            ComponentChangeService.ComponentAdded += ComponentChangeSvc_ComponentAdded;
+            ComponentChangeService.ComponentRemoving += ComponentChangeSvc_ComponentRemoving;
+            ComponentChangeService.ComponentRemoved += ComponentChangeSvc_ComponentRemoved;
+            ComponentChangeService.ComponentChanged += ComponentChangeSvc_ComponentChanged;
         }
 
         // initialize new Manager For Editing ToolStrips
         _editManager = new ToolStripEditorManager(component);
 
+        _host = GetService<IDesignerHost>();
```

Comment:
Use GetRequiredService, designer can't do much work if it doesn't have the host.
```suggestion
        _host = GetRequiredService<IDesignerHost>();
```